rubix-spi/src/main/java/com/qubole/rubix/spi/ClusterManager.java

```diff
@@ -136,11 +155,11 @@ public void initialize(Configuration conf)
     }
   }
 
-  public int getNodeIndex(int numNodes, String key)
+  public String locateKey(String key)
   {
-    HashFunction hf = Hashing.md5();
-    HashCode hc = hf.hashString(key, Charsets.UTF_8);
-    return Hashing.consistentHash(hc, numNodes);
+    String nodeAddress = consistentHashRing.locate(key).orElseThrow(() -> new RuntimeException("Unable to locate key: " + key)).getKey();
+    log.debug("Located key: " + key + " present on node: " + nodeAddress);
```

Comment:
Remove this debug log, this is heavily used method.

Reply:
fixed.